[Bug]: Cannot build Blackmagic Video Output plugin on Windows

### What happened?

When building OpenRV with Blackmagic Video Output support on Windows, one gets the following error:
C1083: Cannot open include file: 'DeckLinkAPI.h': No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\DeckLinkProfileCallback.cpp) [C:\orv\OpenRV\_build\src\plugins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
etc...

This occurs even with Blackmagic_DeckLink_SDK_14.1.zip
rvcfg -DRV_DEPS_BMD_DECKLINK_SDK_ZIP_PATH='/c/Users/labergb/Downloads/Blackmagic_DeckLink_SDK_14.1.zip'


### List all the operating systems versions where this is happening

Windows 11

### On what computer hardware is this happening?

Any

### Relevant console log output

```shell
"C:\orv\OpenRV\_build\src\plugins\output\BlackMagicDevices\BlackMagicDevices.vcxproj" (default target) (139) ->
       (ClCompile target) ->
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicDevices/DeckLinkProfileCallback.h(9,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPI.h': No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\DeckLinkProfileCallback.cpp) [C:\orv\OpenRV\_
       build\src\plugins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicDevices/HDRVideoFrame.h(46,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPI.h'
       : No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\HDRVideoFrame.cpp) [C:\orv\OpenRV\_build\src\plugins\o
       utput\BlackMagicDevices\BlackMagicDevices.vcxproj]
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicDevices/StereoVideoFrame.h(47,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPI
       .h': No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\StereoVideoFrame.cpp) [C:\orv\OpenRV\_build\src\plu
       gins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\DeckLinkSDKAPIDispatch.cpp(7,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPIDispatch.cp
       p': No such file or directory [C:\orv\OpenRV\_build\src\plugins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicDevices/BlackMagicModule.h(11,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPI
       .h': No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicModule.cpp) [C:\orv\OpenRV\_build\src\plu
       gins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicDevices/BlackMagicModule.h(11,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPI
       .h': No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicOutput.cpp) [C:\orv\OpenRV\_build\src\plu
       gins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
         C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\BlackMagicDevices/HDRVideoFrame.h(46,10): fatal  error C1083: Cannot open include file: 'DeckLinkAPI.h'
       : No such file or directory (compiling source file C:\orv\OpenRV\src\plugins\output\BlackMagicDevices\DeckLinkVideoDevice.cpp) [C:\orv\OpenRV\_build\src\plu
       gins\output\BlackMagicDevices\BlackMagicDevices.vcxproj]
```


### Environment variables



### Extra information

rvcfg -DRV_DEPS_BMD_DECKLINK_SDK_ZIP_PATH='/c/Users/labergb/Downloads/Blackmagic_DeckLink_SDK_14.1.zip'

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/cmake/dependencies/bmd.cmake
+++ b/cmake/dependencies/bmd.cmake
@@ -59,6 +59,16 @@ EXTERNALPROJECT_ADD(
   INSTALL_COMMAND ""
 )
 
+# Generate the DeckLinkAPI.h file from DeckLinkAPI.idl provided with the DeckLink SDK
+IF(RV_TARGET_WINDOWS)
+  EXTERNALPROJECT_ADD_STEP(
+    ${_target} post_install_step
+    COMMAND midl.exe ARGS /header DeckLinkAPI.h /iid DeckLinkAPIDispatch.cpp DeckLinkAPI.idl
+    WORKING_DIRECTORY ${_include_dir}
+    DEPENDEES install
+  )
+ENDIF()
+
 ADD_LIBRARY(BlackmagicDeckLinkSDK INTERFACE)
 ADD_DEPENDENCIES(BlackmagicDeckLinkSDK ${_target})
 TARGET_INCLUDE_DIRECTORIES(

